src/components/BookList.spec.js

```diff
@@ -96,4 +102,42 @@ describe('Book list page', () => {
             expect(history.location.pathname).toEqual('/books/abc123');
         });
     });
+
+    describe('Dynamo response has an LastEvaluatedKey indicating more records are available', () => {
+        let rendered;
+
+        beforeEach(() => {
+            documentClient.scan.mockReturnValueOnce({
+                async promise() {
+                    await wait();
+                    return {
+                        Items: [
+                            {
+                                bookId: 'abc123',
+                                title: 'First page Book'
+                            }
+                        ],
+                        LastEvaluatedKey: {foo: 'bar'}
+                    };
+                }
+            });
+            rendered = render(<BookList {...props} />, {wrapper: MemoryRouter});
+        });
+
+        it('should show the books that have loaded', () => wait(() => {
+            expect(rendered.container).toHaveTextContent('A Great Project');
+        }));
+
+        it('should fetch the next page of books', () => wait( () => {
+            expect(documentClient.scan).toHaveBeenCalledWith({
```

Comment:
This feels like testing implementation details rather than what the user would experience. Could we add a test that the rendered page contains entries from two mocked return values, with the first mocked result containing the LastEvaluatedKey?

Reply:
Because we are mocking the documentClient, it is not part of the system under test so I feel like we need to test all of the interactions with it. We could write a mockImplementation that conditionally checks what it is called with and returns either the first page or the second. If the wrong parameters were passed to it it would simply not return the expected data and the test would fail. However, the failure message would be somewhat confusing at it wouldn't identify the exact problem (not passing the correct params to dynamo).

Adding a test case for showing the second page of results is a good idea. I'll add that, then merge.